scripts/generate_can.py

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/env python3
+'''
+Script that generates the DBC file and C source code given the yaml CAN description.
+This script should be called in the root directory of the uwrt_mars_rover_hw_bridge repository.
+'''
+
+import subprocess
+import os
+import cantools
+import yaml
+import shutil
+import re
+
+import generate_can_enums
+import generate_can_wrapper
+
+HWBRIDGE_ROOT_PATH = os.getcwd()
+
+ROVER_CAN_NAME = 'uwrt_mars_rover_can'
+
+YAML_FILE_NAME = ROVER_CAN_NAME + '.yaml'
+DBC_FILE_NAME = ROVER_CAN_NAME + '.dbc'
+DBC_DUMP_FILE_NAME = ROVER_CAN_NAME + '_dump.txt'
+CAN_ENUMS_HEADER_FILE_NAME = ROVER_CAN_NAME + '_enums.h'
+CAN_WRAPPER_HEADER_FILE_NAME = ROVER_CAN_NAME + '_wrapper.h'
+CAN_WRAPPER_SOURCE_FILE_NAME = ROVER_CAN_NAME + '_wrapper.cpp'
+
+SCRIPT_FOLDER_PATH = os.path.join(HWBRIDGE_ROOT_PATH, 'scripts')
+CAN_FOLDER_PATH = os.path.join(HWBRIDGE_ROOT_PATH, 'can')
+GENERATED_FOLDER_PATH = os.path.join(CAN_FOLDER_PATH, 'generated')
+YAML_FILE_PATH = os.path.join(CAN_FOLDER_PATH, YAML_FILE_NAME)
+
+
+AUTOGEN_message_enums = {}
+AUTOGEN_signal_enums = []
+AUTOGEN_msg_map = {}
+AUTOGEN_original_msg_names = {}
+
+
+# https://github.com/eerimoq/cantools/blob/871581b57785fdbd79c118878f9b9c148984963c/cantools/database/can/c_source.py#L762
```

Comment:
I haven't migrated to using a cantools git submodule yet, so just using the cantools python package right now. This camel_t_snake_case conversion was copied from their source code

Reply:
i think the version of cantools needs to be version controlled. submoduled fork if you're modifying it or requirements.txt if you're just using the package. the generated code can change from version to version and we need to make sure to have a concrete way of selecting which version